test/widgets/actions_test.dart

```diff
@@ -0,0 +1,284 @@
+import 'dart:convert';
+
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart' as http;
+import 'package:zulip/api/model/initial_snapshot.dart';
+import 'package:zulip/api/model/narrow.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/model/localizations.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/actions.dart';
+import 'package:zulip/widgets/store.dart';
+import 'package:zulip/widgets/theme.dart';
+
+import '../api/fake_api.dart';
+import '../example_data.dart' as eg;
+import '../model/binding.dart';
+import '../model/unreads_checks.dart';
+import '../stdlib_checks.dart';
+import 'dialog_checks.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('markNarrowAsRead', () {
+    late PerAccountStore store;
+    late FakeApiConnection connection;
+    late BuildContext context;
+
+    Future<void> prepare(WidgetTester tester, {
+      UnreadMessagesSnapshot? unreadMsgs,
+    }) async {
+      addTearDown(testBinding.reset);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        unreadMsgs: unreadMsgs));
+      store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      connection = store.connection as FakeApiConnection;
+
+      await tester.pumpWidget(Builder(builder: (context) =>
+        MaterialApp(
+          theme: zulipThemeData(context),
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
```

Comment:
This makes another spot that could benefit from TestZulipApp, introduced in #805. So whichever of these PRs lands first, the other can update this spot after rebasing.